src/sentry/deletions/base.py

```diff
@@ -126,9 +126,8 @@ def delete_children(self, relations):
                 task=relation.task,
                 **relation.params
             )
-            has_more = task.chunk()
-            if has_more:
-                return has_more
+            while has_more:
```

Comment:
This is only going to run on the first loop iteration, all others will enter this block with `has_more` being set to `False` and this won't delete data from the remaining relations

Reply:
I'm good at programming.

Reply:
Fixed to be loop-local in https://github.com/getsentry/sentry/pull/9664/commits/9e7da76783d3f6781057738b9b4d84900454b6d3.

Reply:
nit: Stylistically, I think the assignment should happen on the line immediately prior to the `while` statement.